Engine/gapi/directx12/dxcommandbuffer.cpp

```diff
@@ -590,6 +590,23 @@ void DxCommandBuffer::dispatch(size_t x, size_t y, size_t z) {
   impl->Dispatch(UINT(x),UINT(y),UINT(z));
   }
 
+void DxCommandBuffer::dispatchIndirect(const AbstractGraphicsApi::Buffer& indirect, size_t offset) {
+  const DxBuffer& ind = reinterpret_cast<const DxBuffer&>(indirect);
+  auto& sign = dev.dispatchIndirectSgn.get();
+
+  curUniforms->ssboBarriers(resState, PipelineStage::S_Compute);
+
+  // block future writers
+  resState.onUavUsage(ind.nonUniqId, NonUniqResId::I_None, PipelineStage::S_Indirect);
+  resState.flush(*this);
+
+  issueExplicitCommonToIndirectStateTransition(ind.impl.get());
```

Comment:
There are 2 issues with this code:
* you do both explicit synchronization and implicit (via `onUavUsage`)
* resource barrier is quite expensive, and calling it per-draw is not a good call